versioned/storage/mongodb/src/main/java/org/projectnessie/versioned/storage/mongodb/MongoDBPersist.java

```diff
@@ -579,26 +581,49 @@ public void deleteObjs(@Nonnull @jakarta.annotation.Nonnull ObjId[] ids) {
   }
 
   @Override
-  public void updateObj(@Nonnull @jakarta.annotation.Nonnull Obj obj)
-      throws ObjTooLargeException, ObjNotFoundException {
+  public void upsertObj(@Nonnull @jakarta.annotation.Nonnull Obj obj) throws ObjTooLargeException {
     ObjId id = obj.id();
     checkArgument(id != null, "Obj to store must have a non-null ID");
 
+    ReplaceOptions options = upsertOptions();
+
     Document doc = objToDoc(obj, false);
-    UpdateResult result = backend.objs().replaceOne(eq(ID_PROPERTY_NAME, idObjDoc(id)), doc);
-    if (result.getMatchedCount() == 0L) {
-      // Testing "modified count" would be wrong here, because it represents the number of actually
-      // changed docs (which isn't the case if the obj is equal to the persisted one).
-      throw new ObjNotFoundException(id);
+    UpdateResult result =
+        backend.objs().replaceOne(eq(ID_PROPERTY_NAME, idObjDoc(id)), doc, options);
+    if (!result.wasAcknowledged()) {
+      throw new RuntimeException("Upsert not acknowledged");
     }
   }
 
+  private static ReplaceOptions upsertOptions() {
+    // A `ReplaceOneModel` with the default replace options (upsert==false) silently does just
+    // nothing.
+    ReplaceOptions options = new ReplaceOptions();
+    options.upsert(true);
+    return options;
+  }
+
   @Override
-  public void updateObjs(@Nonnull @jakarta.annotation.Nonnull Obj[] objs)
-      throws ObjTooLargeException, ObjNotFoundException {
-    // MongoDB does not support bulk conditional updates
+  public void upsertObjs(@Nonnull @jakarta.annotation.Nonnull Obj[] objs)
+      throws ObjTooLargeException {
+    ReplaceOptions options = upsertOptions();
+
+    List<WriteModel<Document>> docs = new ArrayList<>(objs.length);
     for (Obj obj : objs) {
-      updateObj(obj);
+      if (obj != null) {
+        ObjId id = obj.id();
```

Comment:
nit: Shall we `checkArgument(id != null, ...)` as in `upsertObj()`?

Reply:
You can't have an `Obj` with a `null` `id`